### PR TITLE
Fix panic when there is no config file

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

Currently there is panic when config-file is missing:

```
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: proxy_components/proxy_ffi/src/basic_ffi_impls.rs [hunk 1]
```diff
@@ -36,7 +36,11 @@ impl Clone for BaseBuffView {
 
 impl BaseBuffView {
     pub fn to_slice(&self) -> &[u8] {
-        unsafe { std::slice::from_raw_parts(self.data as *const _, self.len as usize) }
+        if self.len == 0 {
```
Comment (CalvinNeo) [inline proxy_components/proxy_ffi/src/basic_ffi_impls.rs:39]: When will len be zero?

Comment (breezewish) [reply proxy_components/proxy_ffi/src/basic_ffi_impls.rs:39]: For example, when we failed to read config (or config file does not exist):

```c++
CppStrWithView GetConfig(EngineStoreServerWrap * server, [[maybe_unused]] uint8_t full)
{
    std::string config_file_path;
    try
    {
        config_file_path = server->tmt->getContext().getConfigRef().getString("config-file");
        std::ifstream stream(config_file_path);
        if (!stream)
            return CppStrWithView{.inner = GenRawCppPtr(), .view = BaseBuffView{nullptr, 0}};
        auto * s = RawCppString::New((std::istreambuf_iterator<char>(stream)),
                                     std::istreambuf_iterator<char>());
        stream.close();
        /** the returned str must be formated as TOML, proxy will parse and show in form of JASON.
         *  curl `http://{status-addr}/config`, got:
         *  {"raftstore-proxy":xxxx,"engine-store":xxx}
         *
         *  if proxy can NOT parse it, return 500 Internal Server Error.
         * */
        return CppStrWithView{.inner = GenRawCppPtr(s, RawCppPtrTypeImpl::String), .view = BaseBuffView{s->data(), s->size()}};
    }
    catch (...)
    {
        return CppStrWithView{.inner = GenRawCppPtr(), .view = BaseBuffView{nullptr, 0}};
    }
}
```

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**